tests/core/pyspec/eth2spec/test/helpers/random.py

```diff
@@ -100,6 +100,13 @@ def set_flag(index, value):
             epoch_participation[index] = flags
 
 
+def randomize_previous_epoch_participation(spec, state, rng=Random(8020)):
+    cached_prepare_state_with_attestations(spec, state)
+    randomize_epoch_participation(spec, state, spec.get_previous_epoch(state), rng)
+    if not is_post_altair(spec):
+        state.current_epoch_attestations = []
```

Comment:
this function sets the previous epoch participation _and also_ clears the current epoch attestations under phase 0 -- it seems like these should be two separate functions but i imagine it is simpler to handle this here than to go and edit a bunch of other tests to handle the situation case-by-case

Reply:
yeah, problem is `cache_prepare_state_with_attestations` is already does a ton of work.

I'm in favor of adding a set of simplified state manipulators for altair+ and slowly pushing out these old ones and only using where necessary